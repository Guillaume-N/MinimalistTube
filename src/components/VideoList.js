import React, { Component } from "react";
import VideoListItem from "./VideoListItem";
import searchVideos from "../YtAPI";

class VideoList extends Component {
  state = {
    videos: []
  };

  componentDidMount() {
    this.term = this.props.match.params.term;
    searchVideos(this.term).then(res => {
      this.setState({ videos: res.items });
    });
  }

  render() {
    if (this.state.videos)
      return (
        <section className="section">
          <div className="container is-centered">
            {this.state.videos.map(video => {
              return <VideoListItem video={video} key={video.etag} />;
            })}
          </div>
        </section>
      );

    return <div>Loading...</div>;
  }
}

export default VideoList;
