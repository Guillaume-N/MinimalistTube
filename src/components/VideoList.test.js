import React from "react";
import { shallow } from "enzyme";
import VideoList from "./VideoList";

describe("VideoList Component", () => {
  it("should match snapshot", () => {
    const mockMatch = {
      params: {
        term: "test"
      }
    };
    const wrapper = shallow(<VideoList match={mockMatch} />);
    expect(wrapper).toMatchSnapshot();
  });
});
