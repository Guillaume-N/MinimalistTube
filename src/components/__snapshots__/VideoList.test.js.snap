// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoList Component should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <VideoList
    match={
      Object {
        "params": Object {
          "term": "test",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="container is-centered"
      />,
      "className": "section",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          Array [],
          false,
        ],
        "className": "container is-centered",
      },
      "ref": null,
      "rendered": Array [
        false,
      ],
      "type": "div",
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="container is-centered"
        />,
        "className": "section",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            Array [],
            false,
          ],
          "className": "container is-centered",
        },
        "ref": null,
        "rendered": Array [
          false,
        ],
        "type": "div",
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
